src/ngraph/runtime/cpu/cpu_debug_tracer.hpp

```diff
@@ -0,0 +1,123 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <fstream>
+#include <functional>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "ngraph/runtime/cpu/cpu_tensor_view_wrapper.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            class CPU_DebugTracer
+            {
+            public:
+                CPU_DebugTracer();
+
+                void set_enable_tracing(bool new_state);
+
+                bool tracing_is_enabled() { return enable_tracing; }
+                void end_of_kernel();
+
+                template <typename T>
+                void dump_one_tensor(const std::string& kernel_name,
+                                     const void* tensor,
+                                     const std::string& tensor_name,
+                                     const size_t size,
+                                     const ngraph::Shape& shape,
+                                     const std::string& in_out);
+
+            private:
+                void init_streams();
+
+                size_t m_serial_number;
+                std::fstream m_tracer_stream;
+                std::fstream m_tracer_bin_stream;
+
+                bool enable_tracing = false;
+            };
+        }
+    }
+}
+
+// use of kahan sum to reduce numeric error
+template <typename T>
+static float find_variance(const std::vector<T>& f_data, float mean, size_t size)
+{
```

Comment:
Don't use `static` here. Leads to warnings in every file that includes this header.